korman/nodes/node_avatar.py

```diff
@@ -71,3 +71,388 @@ def export(self, exporter, bo, so):
+
+        if seek_socket.is_linked:
+            smartSeek = True
+            seek_target = seek_socket.links[0].from_node.target
+        elif seek_socket.auto_target:
+            smartSeek = True
+            seek_target = bo
+        else:
+            smartSeek = False
+            seek_target = bo
+
+        # We need to make sure the object we're attached to is being exported as a Plasma Object, otherwise this is all nonsense
+        seek_target.plasma_object.enabled = True
+
+        if smartSeek:
+            spm = exporter.mgr.find_create_object(plSeekPointMod, name=seek_target.name, bl=seek_target)
+            spm.name = seek_target.name
+
+        msbmod = self._find_create_object(plMultistageBehMod, exporter, bl=seek_target)
+        msbmod.smartSeek = smartSeek
+        msbmod.freezePhys = self.freeze_phys
+        msbmod.reverseFBControlsOnRelease = self.reverse_control
+
+        for stage in self.find_inputs("stage_refs"):
+            animstage = plAnimStage()
+            animstage.animName = stage.anim_name
+            if stage.loop_option == "kLoopForever":
+                animstage.loops = -1
+            elif stage.loop_option == "kLoop":
+                animstage.loops = stage.num_loops
+
+            # Harvest additional AnimStage Settings, if available
+            settings = stage.find_input("stage_settings")
+            if settings:
+                animstage.forwardType = getattr(plAnimStage, settings.forward)
+                animstage.backType =getattr(plAnimStage, settings.backward)
+                animstage.advanceType = getattr(plAnimStage, settings.stage_advance)
+                animstage.regressType = getattr(plAnimStage, settings.stage_regress)
+                for flag in settings.notify_on:
+                    animstage.notify |= getattr(plAnimStage, flag)
+
+                advance_to = settings.find_input_socket("advance_to")
+                if advance_to.is_linked:
+                    # Auto-Advance to specific stage
+                    animstage.advanceTo = advance_to.links[0].from_node.stage_id
+                elif advance_to.auto_advance:
+                    # Auto-Advance
+                    animstage.advanceTo = None
+                else:
+                    # Don't Auto-Advance, just stop!
+                    animstage.advanceTo = -1
+
+                regress_to = settings.find_input_socket("regress_to")
+                if regress_to.is_linked:
+                    # Auto-Regress to specific stage
+                    animstage.regressTo = regress_to.links[0].from_node.stage_id
+                elif regress_to.auto_regress:
+                    # Auto-Regress
+                    animstage.regressTo = None
+                else:
+                    # Don't Auto-Regress, just stop!
+                    animstage.regressTo = -1
+
+            msbmod.addStage(animstage)
+
+    @property
+    def requires_actor(self):
+        return True
```

Comment:
This will cause a CI to be exported for the MSB even when the seek point not the object the tree is attached to. Additionally, a CI is not requested for an explicit seek point. This will generally work because I expect empties will be used, but sometimes you get people like me who use a mesh for these things...

```python
    @property
    def requires_actor(self):
        return not self.find_input_socket("seek_target").is_linked

    @property
    def harvest_actors(self):
        seek_socket = self.find_input_socket("seek_target")
        if seek_socket.is_linked:
            yield seek_socket.links[0].from_node.target.name

Reply:
Excellent, thanks!  I was unsure about this part.

Reply:
`TypeError: 'generator' object is not callable`

Reply:
hwat?

Reply:
I see now. I updated the snippet.